scala3doc/src/main/scala/dotty/dokka/tasty/TypesSupport.scala

```diff
@@ -0,0 +1,260 @@
+package dotty.dokka.tasty
+
+import org.jetbrains.dokka.model._
+import org.jetbrains.dokka.model.{Projection => JProjection}
+import collection.JavaConverters._
+
+trait TypesSupport:
+    self: TastyParser =>
+    import reflect._
+
+    def getGivenInstance(method: DefDef): Option[Bound] = {
+      def extractTypeSymbol(t: Tree): Option[Symbol] = t match
+        case tpeTree: TypeTree =>
+          inner(tpeTree.tpe)
+        case other => None
+
+      def inner(tpe: Type): Option[Symbol] = tpe match
+        case ThisType(tpe) => inner(tpe)
+        case AnnotatedType(tpe, _) => inner(tpe)
+        case AppliedType(tpe, _) => inner(tpe)
+        case tp @ TermRef(qual, typeName) =>
+          qual match
+            case _: Type | _: NoPrefix => Some(tp.termSymbol)
```

Comment:
```suggestion
            case _: TypeRepr | _: NoPrefix => Some(tp.termSymbol)
```

Need to rebase and rename reflection `Type` to `TypeRepr`

Reply:
Done in dd285bbf49f04a8fac15d3aec8cc442e42a7c73e in #10092